syntax = "proto3";

package message_queue;

message Message {
  string queue_name = 1;
}

message GetMessageResponse {
  int32 status = 1;
  string message = 2;
  string username = 3;
  string api_key = 4;
  string service = 5;
  string queue_name = 6;
}

message SendMessageRequest {
  string username = 1;
  string api_key = 2;
  string service = 3;
  string response = 4;
  string queue_name = 5;
}

message SendMessageResponse {
  int32 status = 1;
  string message = 2;
}

service MessageQueue {
  rpc getMessage(Message) returns (GetMessageResponse);
  rpc sendMessage(SendMessageRequest) returns (SendMessageResponse);
}